src/components/Card/Card.tsx

```diff
@@ -1,14 +1,21 @@
-import React, { ReactNode, ReactElement } from "react";
+import React, { MouseEventHandler, ReactNode, ReactElement } from "react";
 import styled, { css } from "styled-components";
 
 export interface CardProps {
   children: ReactNode;
-  withHover?: boolean;
+  onClick?: MouseEventHandler | null;
 }
 
-export const Card = ({ children, ...props }: CardProps): ReactElement => (
-  <StyledCard {...props}>{children}</StyledCard>
-);
+export const Card = ({
+  children,
+  onClick,
+  ...props
+}: CardProps): ReactElement => {
+  if (onClick) {
+    return <StyledCard onClick={onClick} {...props}>{children}</StyledCard>;
+  }
+  return <StyledCard {...props}>{children}</StyledCard>;
```

Comment:
I think we can just do

```suggestion
  return <StyledCard {...props}>{children}</StyledCard>;
```

and not destructure the `onClick` from props